### DISCUSSION TITLE
Building done, how to run?

### DISCUSSION THREAD

**Discussion by ZayaPool** [mtconnect/cppagent#414] [Q&A]
Hello
After a lot of work, (I have little experience with computer programming) I successful built the agent on Ubuntu 20.04.3. All the test done after the building were successfull and I dont have any error messages, however, I dont know how to launch the agent, I dont find any directory with /bin. I searched on /home/zaya/cppagent/test_package/build/gcc11x86_64-gnu17-release and there is just a bunch of files. I also search on home/.conan2/p/b... but I didnt find the bin directory. 

My objective is to use this agent in another linux computer: a raspberry pi3+. (It was impossible to build the agent in this raspberry since it kept asking me for cmake version >3.23 and I found multiple errors when I tried to compile this version of cmake from the source code, so I gave up and I am trying to build it on my desktop ubuntu and then transfer it somehow to the raspbian OS)

Thank you in advance for your help!

**Comment by robot-ranger** [comment]
is it in /usr/bin/ ?

we use the docker image, it has been a while since we have built it ourselves.
once you find the executable, you launch it with `mtcagent run /path/to/your/agent/cfg`

Consider using the docker images. you wont have to build anything; its automatically built on github and pushed to docker hub here:
https://hub.docker.com/r/mtconnect/agent

**Comment by wsobel** [comment]
The agent is installed in /usr/bin which is the non-system bin directory.

The paths for search (config can be specified as the third arg with full path) can be given in the Agent.cfg as paths:

- `ConfigPath = path1,path2,path3,...`
- `DataPath= path1,path2,path3,...`
- `WorkingDirectory = dir`
- `PluginPath = path1,path2,path3,...`

These will be the default search paths for the files in the configuration file like schemas will search the data path, etc.

For docker the agent files are installed in the following directories:

* `/mtconnect/data` for styles and schemas
* `/mtconnect/config` for the `agent.cfg` and `Devices.xml`
* `/mtconnect/log` for log files

We are following the same pattern as `mosquitto`
